lib/rubocop/cop/lint/block_alignment.rb

```diff
@@ -148,12 +148,18 @@ def block_is_on_next_line?(begin_node, block_node)
           begin_node.loc.line != block_node.loc.line
         end
 
+        def ancestor_on_same_line(node)
+          node.ancestors.reverse.find do |ancestor|
+            ancestor.loc.line == node.loc.line
+          end
+        end
+
         def autocorrect(node)
-          key = node.children.first
+          ancestor_node = ancestor_on_same_line(node)
```

Comment:
Aaah.. that's a good point, thanks. I started to interpret your comment totally wrong :smile:. I'll fix that along with a new guard clause.